### PR TITLE
Add browser support

### PR DESCRIPTION
**Pull request by riknoll**
This PR splits pxt-mkc into three packages:

1. mkc-core, which contains the shared code for the other two packages and has no node dependencies
2. mkc-node, which contains the node CLI
3. mkc-browser, which contains a browser implementation of the language service

mkc-node will be published as the existing makecode package. The other two are new packages. The CLI should work exactly the same as before.

Opening as a draft PR because converting this repo into a monorepo broke a few things (e.g. the build, semantic-release)

### CODE REVIEW

File: packages/mkc-browser/package.json [hunk 1]
```diff
@@ -0,0 +1,70 @@
+{
+  "name": "mkc-browser",
```
Comment (pelikhan) [inline packages/mkc-browser/package.json:2]: there is already an unrealated "mkc" package in npm, so we should stay away from this name

Comment (pelikhan) [reply packages/mkc-browser/package.json:2]: https://www.npmjs.com/package/mkc

File: packages/mkc-browser/package.json [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    "arcade",
+    "makecode"
+  ],
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/microsoft/pxt-mkc"
+  },
+  "author": "",
+  "license": "MIT",
+  "homepage": "https://makecode.com/docs",
```
Comment (pelikhan) [inline packages/mkc-browser/package.json:19]: probably point to something more precise

File: packages/mkc-browser/package.json [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+  "files": [
+    "README.md",
+    "built/*",
+    "external/*",
+    "makecode"
+  ],
+  "preferGlobal": false,
+  "engines": {
+    "node": ">= 11.0.0"
```
Comment (pelikhan) [inline packages/mkc-browser/package.json:28]: should we bump this to 14 at least?

File: packages/mkc/package.json [hunk 4]
```diff
@@ -0,0 +1,72 @@
+{
+  "name": "mkc-core",
```
Comment (pelikhan) [inline packages/mkc/package.json:2]: same as mkc-browser

**Comment by pelikhan** [comment]
Can you update the local development section of the README with information about various package; and/or update build instructions if needed.

**Review by pelikhan — APPROVED** [review]

File: .gitignore [hunk 5]
```diff
@@ -5,3 +5,5 @@ node_modules
 pxt_modules
 *.vsix
 src/simloaderfiles.ts
+packages/mkc/src/simloaderfiles.ts
+packages/mkc-browser/src/workerFiles.ts
```
Comment (mmoskal) [inline .gitignore:9]: makecode-browser not mkc-browser?

File: packages/makecode-browser/src/workerFiles.ts [hunk 6]
```diff
@@ -0,0 +1,178 @@
+export const workerJs = `
```
Comment (mmoskal) [inline packages/makecode-browser/src/workerFiles.ts:1]: was this ignored?

**Comment by riknoll** [comment]
alright @pelikhan, I updated the packages to all be `makecode-*` instead of mkc and fixed the build/updated the readme. I also added semantic-release-monorepo which *should* fix semantic-release though I obviously haven't tested that yet.

Take another look real quick?

File: packages/makecode-core/src/simloaderfiles.ts [hunk 7]
```diff
@@ -0,0 +1,357 @@
+export const simloaderFiles: Record<string, string> = {
```
Comment (mmoskal) [inline packages/makecode-core/src/simloaderfiles.ts:1]: should be ignored?

**Review by mmoskal — APPROVED** [review]
looks good!

**Review by pelikhan — APPROVED** [review]

**Pull request merged**